auslib/admin/views/base.py

```diff
@@ -45,27 +45,38 @@ def decorated(*args, **kwargs):
     return wrap
 
 
+def catchOutDataError(messages):
+    def wrap(f):
+        def decorated(*args, **kwargs):
+            try:
+                return f(*args, **kwargs)
+            except OutdatedDataError as e:
+                msg = "Couldn't preform the request %s. Outdated Data Version. old_data_version doesn't match current data_version: %s" % (messages, e)
+                cef_event("Bad input", CEF_WARN, errors=msg)
+                return Response(status=400, response=json.dumps({"data": e.args}))
+        return decorated
+    return wrap
+
+
 class AdminView(MethodView):
 
     def __init__(self, *args, **kwargs):
         self.log = logging.getLogger(self.__class__.__name__)
         MethodView.__init__(self, *args, **kwargs)
 
+    @catchOutDataError("POST")
     def post(self, *args, **kwargs):
         self.log.debug("processing POST request to %s" % request.path)
         with dbo.begin() as trans:
             return self._post(*args, transaction=trans, **kwargs)
 
+    @catchOutDataError("Update")
```

Comment:
You're using an HTTP method name above ("POST"), and then switching to a non-HTTP name here and for delete(). I'm fine with either, but please be consistent.